docs/tutorial/t-spark-shell.md

```diff
@@ -0,0 +1,144 @@
+"""
+```
+
+The following is a function that returns the number of vowel characters in the string:
+
+```python
+def count_vowels(text: str) -> int:
+  count = 0
+  for char in text:
+    if char.lower() in "aeiou":
+      count += 1
+  return count
+```
+
+To test this function, the string `lines` can now be passed into it and the number of vowels is printed to the console as follows:
+
+```python
+>>> count_vowels(lines)
+128
+```
+
+Since Spark is a distributed processing framework, we can split up this task and parallellize it over multiple executor pods. This parallelization can be done as simply as:
+
+```python
+>>> from operator import add
+>>> spark.sparkContext.parallelize(lines.splitlines(), 2).map(count_vowels).reduce(add)
+128
+```
+
+Here, we parallelize the the tax into two executors (passed as an argument to `parallelize` function), where each line is processed by an executor pod. The number of vowels in each line is then added up to calculate the total number of occurrences of vowel characters in the string. This kind of parallelization of task is particularly useful in processing very large data sets which helps in reducing the processing time significantly.
+
+To exit from PySpark shell, you can simply run `exit()` or press `Ctrl` + Z key combination.
+
+
+### Scala Shell
+Spark comes with a built-in interactive Scala shell as well. Enter the following command to enter an interactive Scala shell:
+
+```bash
+spark-client.spark-shell --username spark --namespace spark
+```
+
+Once the shell is open and ready, you should see a prompt simlar to the following:
+
+```
+Welcome to
+      ____              __
+     / __/__  ___ _____/ /__
+    _\ \/ _ \/ _ `/ __/  '_/
+   /___/ .__/\_,_/_/ /_/\_\   version 3.4.1
+      /_/
+         
+Using Scala version 2.12.17 (OpenJDK 64-Bit Server VM, Java 11.0.20.1)
+Type in expressions to have them evaluated.
+Type :help for more information.
+
+scala> 
+```
+
+Just as in PySpark shell, the spark context and spark session are readily available in the shell as `sc` and `spark` respectively. Moreover, new executor pods are created in `spark` namespace in order to execute the commands, just like in the PySpark shell.
+
+The example of counting the vowel characters can be equivalently run in Scala with the following lines:
+
+```scala
+val lines = """Canonical's Charmed Data Platform solution for Apache Spark runs Spark jobs on your Kubernetes cluster.
+You can get started right away with MicroK8s - the mightiest tiny Kubernetes distro around! 
+The spark-client snap simplifies the setup to run Spark jobs against your Kubernetes cluster. 
+Spark on Kubernetes is a complex environment with many moving parts.
+Sometimes, small mistakes can take a lot of time to debug and figure out.
+"""
+
+def countVowels(text: String): Int = {
+  text.toLowerCase.count("aeiou".contains(_))
+}
+
+sc.parallelize(lines.split("\n"), 2).map(countVowels).reduce(_ + _)
```

Comment:
```suggestion
scala> val lines = """Canonical's Charmed Data Platform solution for Apache Spark runs Spark jobs on your Kubernetes cluster.
You can get started right away with MicroK8s - the mightiest tiny Kubernetes distro around! 
The spark-client snap simplifies the setup to run Spark jobs against your Kubernetes cluster. 
Spark on Kubernetes is a complex environment with many moving parts.
Sometimes, small mistakes can take a lot of time to debug and figure out.
"""

scala> def countVowels(text: String): Int = {
  text.toLowerCase.count("aeiou".contains(_))
}

scala> sc.parallelize(lines.split("\n"), 2).map(countVowels).reduce(_ + _)
```
For clarity and formatting consistency (no prompt for standard shell, prompt for special shells)

Reply:
Wouldn't this make copying the commands quite difficult though? One has to copy them one by one, and if someone copies the entire block (using the copy button on top right), the copied command would not work (since there is extra string `scala>` in the beginning.

Reply:
Yes, and that's a valid point that often gets discussed during style guide debates about command prompts. 
But then we'd have to keep it consistent by not including the python shell prompt either (`>>>`), and making sure it's clear which shell we're talking about, since there are three involved.